File: 1-TwoSum/1-TwoSum.java
// Last updated: 6/13/2026, 9:10:32 PM
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        HashMap<Integer, Integer> hm = new HashMap<>();
4        for (int i = 0; i < nums.length; i++) {
5            int complement = target - nums[i];
6            if (hm.containsKey(complement)) {
7                return new int[] { hm.get(complement), i };
8            }
9            hm.put(nums[i], i);
10        }
11        return new int[] {};
12    }
13}

File: 1-TwoSum/1-TwoSum.c
// Last updated: 5/26/2026, 12:35:08 PM
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int *result = (int *)malloc(2 * sizeof(int)); // Allocate for 2 ints
    *returnSize = 0;

    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                result[0] = i;   
                result[1] = j;    
                *returnSize = 2;
                return result; 
            }
        }
    }
    return result; 
}